### ISSUE TITLE
Auto-close enhancements

### ISSUE THREAD

**Issue by herbat** [processing/processing#5032]
I see that auto-close has the basic expected behavior, but it would be a lot easier if 
1. inserting `{` would also insert newline and indent
2. if an auto-closed character( `( { [ ' "` ) and its pair are *both* typed by the user, delete the inserted closing character
3. deleting the auto-closed character will delete it's pair as well as its closing pair we just inserted

I know these are minor improvements, but if I could find the place the current auto-close functionality is located, I would be happy to code it. Any other recommendations/ideas welcome!
Thanks for any help!

**Comment by GKFX** [comment]
I disagree with case 1: code such as `int[] x = {1, 2, 3};` one one line is fine, and it's deliberately left alone by the autoformatter on `ctrl`+`T`. It would be made much harder to write if Processing automatically added a bunch of newlines to it. Otherwise multidimensional arrays get an unnecessary number of newlines.
If you're looking for where to start reading source files, try looking under https://github.com/processing/processing/tree/master/app/src/processing/app/syntax, which is the text area component, although I can't confirm that the auto-closing is in there.
